src/main/java/io/vavr/control/Either.java

```diff
@@ -483,6 +483,50 @@ public final void orElseRun(Consumer<? super L> action) {
         }
     }
 
+    /**
+     * Calls recoverFunction if the projected Either is a Left, performs no operation if this is a Right. This is
+     * similar to {@code getOrElseGet} where the fallback method also returns an Either.
+     *
+     * <pre>{@code
+     * Either<Integer, String> tryGetString() { return Either.left(1); }
+     *
+     * Either<Integer, String> tryGetStringAnotherWay(Integer lvalue) { return Either.right("yo " + lvalue); }
+     *
+     * = Right("yo 1")
+     * tryGetString().recover(this::tryGetStringAnotherWay);
+     * }</pre>
+     *
+     * @param recoverFunction a function which accepts a Left value and returns an Either
+     * @return an {@code Either<L, R>} instance
+     */
+    public final Either<L, R> recover(Function<L, Either<L, R>> recoverFunction) {
```

Comment:
Please add `Objects.requireNonNull(recoverFunction, "recoverFunction is null");` and a `@throws NullPointerException if the given {@code recoverFunction} is null`.

Reply:
2 more things:

### 1) Please rename `recover` to `recoverWith`

This is by name convention (taken from Scala and also VAVR's Try.recover* functions) because an `Either<L, R>` is returned instead of an `R` by the recovery function.

### 2) Please add generic type bounds

* _contra-variant_ generic type bound `? super L` for the _consumer_ (aka domain) of the recover function
* _covariant_ generic type bound `? extends Either<? extends L, ? extends R>>` for the _producer_ (aka codomain) of the recover function

Result:

```java
Either<L, R> recoverWith(Function<? super L, ? extends Either<? extends L, ? extends R>> recoverFunction)
```

Reply:
Done